src/client/io_loop.rs

```diff
@@ -937,6 +937,24 @@ impl<T: InvokeUiSession> Remote<T> {
         }
     }
 
+    async fn send_toggle_virtual_display_msg(&self, peer: &mut Stream) {
+        let lc = self.handler.lc.read().unwrap();
+        let displays = lc.get_option("virtual-display");
+        for d in displays.split(',') {
+            if let Ok(index) = d.parse::<i32>() {
+                let mut misc = Misc::new();
+                misc.set_toggle_virtual_display(ToggleVirtualDisplay {
+                    display: index,
```

Comment:
No size? Have you confirmed we have exactly the same behavior as ToDesk?

Reply:
> No size (resolution)? Can we restore resolution after connect next time?

Yes. The resolutions in system `Display settings` (`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\GraphicsDrivers\Connectivity`) are used.

> Have you confirmed we have exactly the same behavior as ToDesk?

Not exactly.

1. ToDesk does not allow multiple control side, it requires external channel plug-in.
2. We can't control the virtual display index. So there is a disadvantage.

![image](https://github.com/user-attachments/assets/fc55595d-15d0-4e21-acd0-229085bb417a)

Reply:
> The resolutions in system `Display settings` (`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\GraphicsDrivers\Connectivity`) are used.

You have unplugged, the resolution info still there?

Reply:
Yes.

The display settings are stored in registry `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\GraphicsDrivers\Connectivity`.
The displays settings will be applied for the same display combinations.

Eg. You have two displays `A`, `B`, you have ever set the display settings like:
1. `A`, `1920x1080`
2. `B`, `1440x900`
3. `A+B`, `2560x1440` and `1024x768`

Then, you have plugged out all displays
1. Plug in `A`, the resolution is `1920x1080`
2. Plug in `B`, the resolutions will be `2560x1440` and `1024x768`
3. Plug out `A`, the resolutions is `1440x900`



For `amyuni` idd, the resolutions are just changed without recording

https://github.com/rustdesk/rustdesk/blob/ce56be65077602d6fa4713ecb8d80d7ce6e71d42/src/server/connection.rs#L2711

Then, the controlled side will never try recovery the resolution set by self.

The plugged in display will always apply the system display settings. (The combination of the displays).